src/System.Private.CoreLib/src/System/Reflection/RtFieldInfo.cs

```diff
@@ -309,10 +273,9 @@ public override Type FieldType
         {
             get
             {
-                if (m_fieldType == null)
-                    m_fieldType = new Signature(this, m_declaringType).FieldType;
+                return m_fieldType ?? Initialize();
 
-                return m_fieldType;
+                Type Initialize() => m_fieldType = new Signature(this, m_declaringType).FieldType;
```

Comment:
This doesn't need to be NoInlining (in which case it also currently can't be local)?